internal/converter/internal/otelcolconvert/converter.go

```diff
@@ -11,9 +11,9 @@ import (
 	"go.opentelemetry.io/collector/otelcol"
 )
 
-// componentConverter represents a converter which converts an OpenTelemetry
+// ComponentConverter represents a converter which converts an OpenTelemetry
 // Collector component into a Flow component.
-type componentConverter interface {
+type ComponentConverter interface {
```

Comment:
nit: If we're going to make this interface public, we should make `state` public too. That would also make it possible to write converters in other packages outside of this one.

Reply:
There were also some util functions I needed but I've implemented it this way.